src/Microsoft.Data.Analysis/DataFrameRow.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Data.Analysis
+{
+    /// <summary>
+    /// A DataFrameRow is a collection of values that represent a row in a <see cref="DataFrame"/>.
+    /// </summary>
+    public class DataFrameRow : IEnumerable<object>
+    {
+        private DataFrame _dataFrame;
+        private long _rowIndex;
+        internal DataFrameRow(DataFrame df, long rowIndex)
+        {
+            _dataFrame = df;
+            _rowIndex = rowIndex;
```

Comment:
Should there be an assert / check that `rowIndex` is within bounds? Or is it OK to handle this later / bail on error?

Reply:
It's an internal constructor, I doubt someone would pass in an invalid rowIndex. I added one any way though.